### ISSUE TITLE
Raspberry Pi build broken

### ISSUE THREAD

**Issue by 01micko** [puppylinux-woof-CE/woof-CE#1]
I just did a sanity check, doing a No-X build, crashes at 3builddistro stage

**Comment by mavrothal** [comment]
Where do we stand on this one?

**Comment by 01micko** [comment]
out to pasture?

**Comment by mavrothal** [comment]
Lovely place to be :D

**Comment by woodenshoe-wi** [comment]
Well... This spring I couldn't resist and I got a Pi2 and over the summer I did succeed in hacking the woof build scripts enough to make a bootable image that would work with either the 512MB model B or the new Pi2.  The Pi2 is a quad core and uses a different kernel, so I modified the scripts to load two kernels into the image.  I had success treating the kernels and firmware like any other package and loading them from raspbian.org with the rest of the packages, but there was a puppy specific program used by the fixmenus script, and the ARM version is an old version ( I think jwm-xdgmenu, but that's just from memory ) that can't parse the config file as it currently is in woof.  I have the modified woof script files somewhere, but haven't gotten around to making the changes into git commits yet, and am kind of busy getting ready for winter outside.

Short answer:
Probably still broken, at least for Pi2, but I know it can be fixed.
(but it may take me a while to get around to it.)

**Comment by woodenshoe-wi** [comment]
I found back some of my notes...
...
xdg_puppy-0.7.6-15-binaries_only-armv6.pet
is too old to handle the "MENHEIGHT" option in
/usr/sbin/fixmenus
...

**Comment by 01micko** [comment]
You can compile the newer version. It is at http://distro.ibiblio.org/puppylinux/sources/x

**Comment by woodenshoe-wi** [comment]
I may be missing something, but the only version I could find was
xdg_puppy-0.7.6-9.tar.bz2
Is this newer than 0.7.6-15?

**Comment by woodenshoe-wi** [comment]
I tried it and it worked!?
I don't understand the version numbering.

**Comment by woodenshoe-wi** [comment]
Assuming you wanted the pet, I uploaded it to Drop Box.
https://www.dropbox.com/s/hmk012wqld98w8f/xdg_puppy_jwm-0.7.6-16-armv6.pet?dl=0
I'm guessing at the version number, but I figured that if it worked it must be newer than 0.7.6-15.

I also used petbuilds to compile JWM.
https://www.dropbox.com/s/l34rhvxmw8ogevu/jwm-2.3.2-armv6l_hf.pet?dl=0

Both of these were compiled in a test woof build based on Raspbian wheezy on my model B Pi.
I think they would go in Packages-puppy-wheezy-official?

I haven't finished making all the changes into commits, but I did finish a few and pushed them up to my git-hub account.  They have only been tested with builds for raspberry pi, would it be OK to issue pull requests or do they need to be tested on an x86 build first?

**Comment by mavrothal** [comment]
The one commit I found in your repo at ( a4808b3 ) appears safe, but is always a good idea to run a test build for x86.

Regarding the pets, why not in Packages-puppy-armv6-official. Is there any compile option that would break them in other flavours (that support hard float of course)?
Or maybe open a Packages-puppy-armv6hf repo?
Not that makes a huge difference but if there are more to come is better to get them together I think.

**Comment by woodenshoe-wi** [comment]
I didn't use any special compile options, but I'm assuming they're dynamically linked against system libs from Raspbian.  Some pets from Packages-puppy-armv6-official didn't run right in my Raspbian based builds so I assumed my pets wouldn't run right in other flavors either.  If it is a problem to compile new pets for each flavor I could try compiling them statically soft float, then they could run on any armv6, but as they are now they may or may not work depending on what library versions the other flavors use.

I'm still tracking down dependencies so I haven't made my changes to DISTRO_PKGS_SPECS-raspbian-wheezy into a commit yet, but I didn't know if you wanted me to dump all my changes on you at once, or as I finished testing them. a4808b3 735b39f 412006c

**Comment by mavrothal** [comment]
Pulls with many commits have the risk to get rejected because of some little issue in one of the commits, but makes more sense to  get them in one pull if they are related as in this case. 
Re-issuing a pull with one commit or many does not make a lot of difference really so I would rather say go with one, but it depends on how comfortable you are with git and pull requests.

**Comment by woodenshoe-wi** [comment]
The three commits I finished should allow a build that will boot to the command line, but JWM won't start and  jwm-xdgmemu can't parse the menu properly.  I'm still tracking down dependencies of dependencies, and if I have to do a x86 test build I would rather wait until I am all the way done.

The two issues that I can't solve myself are JWM and xdg_puppy.
One option is to use the dynamically linked pets in Drop Box,
the other option is to try to compile xdg_puppy statically against uclibc 
and take the files from petbuilds and make a template to use JWM from Raspbian.

Until this is figured out, my "working" builds won't be reproducible by others.

**Comment by woodenshoe-wi** [comment]
I did a test build last night with just the commits in
https://github.com/woodenshoe-wi/woof-CE/tree/fix-pi-build
and the problems with JWM and xdgmenu somehow went away by themselves.

I still need to do a x86 test build, but I think I am close to being finished.

**Comment by mavrothal** [comment]
:+1:

**Comment by woodenshoe-wi** [comment]
Well, I've been running into snags with my x86 build.  First I tried the slacko 14.1 build running woof-CE in precise but ran into problems in 2createpackages, so I installed the latest slacko on my parent's computer and then 2createpackages worked but when I tried to run 3builddistro it said I had to run 3builddistro-Z instead which is not the script I edited so that won't work for a test.

I also was working on getting rpi-update to work and I think I have to add one more line to fstab for /boot which is done in 3builddistro I think.

I want to try to finish the fix for rpi-update before I try another x86 build, but I can't recommend actually running rpi-update because I think it breaks the arrow keys on my keyboard.

I guess I'm not as close to being finished as I hoped :-(

**Comment by mavrothal** [comment]
> I guess I'm not as close to being finished as I hoped :-(

You are closer than anybody else :+1:

**Comment by 01micko** [comment]
I ordered the new 'el cheapo' pi. Still set me back AUD$32.50 :scream: but it does come with hdmi adapter and USB adapter. We should be able to make use of FatDogARM repo by @jamesbond3142 :wink:

**Comment by dimkr** [comment]
It's ARMv6 like Raspberry Pi 1 :sleeping:

**Comment by 01micko** [comment]
Doesn't matter.. we can build up a repo.. :+1: petbuilds should help.

**Comment by mavrothal** [comment]
But et's have something taht builds first :wink:

**Comment by 01micko** [comment]
TBH I don't like the idea of downloading an image. It should be created on the fly with builder's choice of ext2, or ext3,4 (no journal)., with builder's choice of size. 

Actually, if you make the small FAT partition, then swap, then the Linux partition it becomes trivial to resize the thing, so a default of fitting on a 2 GB media is easy to resize (with gparted or tool of choice).

**Comment by woodenshoe-wi** [comment]
For those of you that have a Pi or are buying one, I have my work uploaded to git-hub and you can start playing with it now if you would like.
https://github.com/woodenshoe-wi/woof-CE.git
If you are running anything older than Slacko 6.3.0 you will need to update debdb2pupdb or you will have errors finding the raspberrypi-bootloader-nokernel package.
See https://github.com/woodenshoe-wi/woof-CE#raspbian-build

On the subject of fixing the the Pi build in woof-CE, I tried a slacko 14.0 build and when I tried to boot the CD on my laptop I got this
![screen-shot](https://cloud.githubusercontent.com/assets/7399935/11521910/027b14e0-9877-11e5-8220-54b4fc171ff6.png)

**Comment by wdlkmpx** [comment]
A lot of changes have happened since the fire nation attacked. I wonder if we can port some of the commits from woodenshoe. I hope his pet pkgs are in the ibiblio arm repo or something

**Comment by woodenshoe-wi** [comment]
I thought I would try merging https://github.com/woodenshoe-wi/woof-CE/tree/fix-pi-build and do a test build for my pi.  It booted and everything seemed to work until I realized that I had merged master and not testing.

When I tried to merge testing instead of master I found that 3builddistro had disappeared...
From 3builddistro-Z in testing ...
```
if [ "$DISTRO_KERNEL_PET" != "Huge_Kernel" ];then
    echo "You can not use this script. Use 3builddistro instead or click the button"
    echo "named 'BUILD DISTRO (normal build)' in the Woof GUI"
    echo "Press enter to exit."
    read normal_build
    exit 0
fi
```
Back when I had been working on this, it seemed that 3builddistro-Z was for "Huge_Kernel" and 3builddistro was for everything else.  Is the 3builddistro-Z script in rationalise going to support alternatives other than "Huge_Kernel"?

I was using the kernels from Raspbian and have no plans to try to compile a kernel on my pi.

**Comment by wdlkmpx** [comment]
You're raising good points, many things have changed while woof was being simplified. I guess someone has to compile a huge kernel or cross-compile.  For arm boards people usually cross compile in a x86 host.

The rationalise branch has many more changes, so i think everything has to be restarted. But we're lacking devs. probably it's all dead now according to a vision i had..

I'm not sure, but this can be the solution
http://autobuild.buildroot.org/toolchains/tarballs/

This stuff runs in x86_64 systems and cross-compiles for armv6 armv7, etc. Assuming everything is properly set up, that's something to be added to the kernel-kit. Something similar to initrd-progs, which uses the tarballs from aboriginal linux i guess...

For this maybe we just have to copy one of the existing kernel configs and tweak it for an arm build, or stole a proper config and just use it, the latter seems to be most reasonable solution.

**Comment by woodenshoe-wi** [comment]
The other option I was going to suggest was to have a separate 3pibuild script like the 4babybuild and 4quirkybuild scripts, but I see that they have been removed too.

I do have a little experience with buildroot, and was trying to make a kind of cross between buildroot and the Puppy Linux idea of a devx sfs, but I don't plan to compile a kernel unless I am compiling everything else from source.  If I am using debs from Raspbian, I want to use the kernel from Raspbian also.

I was also able to use rpi-update https://github.com/Hexxeh/rpi-update to update the kernel and firmware from a running pi.  I don't know how normal it is to compile a kernel for a pi yourself.

Maybe someone could pack up the Raspbian or github kernel into the format expected for a "Huge_Kernel", but it would not stay up to date automatically.

**Comment by wdlkmpx** [comment]
Good points, where can i download a pi kernel, just to see. Maybe i can do something. Although you will encounter new problems as the old .pet kernels stuff doesn't exist anymore.

Well, i think you're right, 3pibuild makes sense actually.

I've never used an arm board, so this is unknown territory for me. And it's unlikely i'll buy one, i read it's slow. i already have and am struggling with slow hardware due my masochistic personality traits.

So, a pi puppy is basically a full install right,,, no sfs's, etc? The there should be no problem using a raspbian kernel..

Well, at least in x86 the kernel is always compiled for puppy, and it works in basically any woofce puppy, the only restriction that can be found is probably when compiling new modules... haven't tried.

A couple weeks ago i was looking at the slackware website, and found they also provide an arm port, but they link to fatdog.eu. I wonder if the fatdog team has anything to do with it. Might be of interest.

http://sarpi.fatdog.eu/index.php?p=downloads

Overall  slackware, arch, gentoo and a few other distros don't split their packages, so everything is cleaner and less confusing.

**Comment by wdlkmpx** [comment]
The build process is being automated so that only a couple of questions are asked, everything else is automatic, most options can be specified through a config file (_00build.conf). Well, for x86 at the moment.

The arm stuff remains in the darkness, but hopefully it will be sorted out starting from 2017 haha

This is where the actions is happening:
https://github.com/puppylinux-woof-CE/woof-CE/commits/rationalise

Everything in that branch will go to testing soon

**Comment by woodenshoe-wi** [comment]
The Raspbian kernels are located at http://archive.raspbian.org/raspbian/pool/main/l/linux-latest/
According to the wheezy package list linux-image-rpi-rpfv_3.18+46+rpi4_armhf.deb is the recommended version for the pi 1 and linux-image-rpi2-rpfv_3.18+46+rpi4_armhf.deb is the recommended version for the pi 2.  I have not tried to do a build with jessie.

A raspberry pi also needs boot-loader / firmware files.

The Raspbian ones are at http://archive.raspbian.org/raspbian/pool/firmware/r/raspberrypi-firmware-nokernel/
According to the wheezy package list raspberrypi-bootloader-nokernel_1.20150212-1~nokernel1_armhf.deb

When I was first getting my pi to boot alpha4 http://murga-linux.com/puppy/viewtopic.php?p=665256#665256 I was grabbing kernels and firmware from https://github.com/Hexxeh/rpi-firmware

I was treating the Raspbian debs as regular packages and the woof scripts would figure out which versions were specified in the package lists, so only the package names had to be specified in DISTRO_PKGS_SPECS-raspbian-wheezy, eg. 
`yes|raspberrypi-bootloader-nokernel|raspberrypi-bootloader-nokernel|doc>null
`
Since a raspberry pi build is a bit different than a normal puppy build, installing two kernels and a bunch of firmware files, using a separate script could just download the packages like I am doing now and not bother with a "Huge_Kernel" package.  If a separate 3pibuild script would be OK I could start with the last 3builddistro script, strip out anything not pi related, and try to implement any _00build.conf features using 3builddistro-Z as a reference?

Currently a raspberry pi build is kind of like a full install.  The kernels and firmware are installed into the first partition of a filesystem image with a FAT filesystem (the boot-loader needs FAT), then there is a swap partition in the image and a linux type partition where everything else is installed.  I avoid loading the swap partition because I am worried it could wear out the SD card.  The Raspbian kernels include squashfs, and a devx sfs is made during the build process that can be loaded on the pi to allow compiling.  The Raspbian kernels also include overlayfs, but I don't know if it is feasible to get puppy to work with overlayfs.

**Comment by wdlkmpx** [comment]
OK, i believe you have the complete idea. so it will be better for you step in and join.

Well, puppy uses aufs, but it doesn't matter when the filesystem is complete, i.e. full install. although the code always assumes a layered filesystem, but works nonetheless (i think).

The initrd init used in the x86 builds always assumes aufs and sfs files. it's for a modular approach and fancy handy stuff.

Overlayfs is the winner indeed - in this battle against aufs. but it's not supported in puppy.

when it comes to packages, i can help get everything ready, i mean editing distro_pkgs_specs and uploading files to the repo so the build works.,. if you have any files ready, just open a pull request or post the files somewhere.

regarding 3pibuild, i guess you can do it, but the whole process was meant to unify eveything, applying the same patches and tweaks for a puppy build. 

in rationalise, the scripts are, let's say, quite small compared to what they used to be and more changes are coming, so i think you might to do your stuff directly in rationalise after it's merged to testing. that branch is for experimental stuff, but once there are number of commits, everything else tends go there, just like it's happening now.

my thinking is try and do something even if that breaks something else (will be fixed), rather than do nothing

**Comment by woodenshoe-wi** [comment]
I looked at the 3builddistro-Z script in rationalise and recognized a lot of it from 3builddistro, I had thought that 3builddistro-Z was a complete rewrite and all of that would be gone.  I will try to make my changes to 3builddistro-Z.

Currently a raspberry pi build does not use an initrd, although I think the pi boot-loader does support one.

Unless there has been a format change, I am familiar with all the distro_pkgs_specs files, and the two pets I mentioned earlier may or may not be needed.  Most of the fixing had to do with finding dependencies and fixing templates, I can't think of anything else I had to compile.

**Comment by wdlkmpx** [comment]
A complete rewrite is in the woof-next branch, it's a different
approach and makes puppy part of the parent distro or something like
that. it discards many things, this is what probably would have been
the right direction as in woofce there is not an agreement to follow a
certain path as i'm seeing in the jwm stuff.

I'm building a precise pup testing the changes i make and i'm quite
familiar with the package dependency in the
ubuntu/debian/devuan/trisquel configs.. because they all follow
debian's scheme. all of them have cdrkit instead of cdrtools, all of
them have libav instead of ffmpeg... the latter choice made compile
ffmpeg and recompile all the apps that use libav. this makes puppy a
truly alien guest in the grand scheme of things.

i just compiled jwm and rox, and removed jwmconfig and jwm themes from
the distro_pkgs_specs... to use the woofce jwm config

i think many old packages can be removed to make the configs look
cleaner and easily maintainable. i deleted about 300 entries and
everything seems more reasonable. and it still has almost all the
stuff.

**Comment by woodenshoe-wi** [comment]
#900 Not polished, but something to start with.

**Comment by wdlkmpx** [comment]
Merged. To make your steps reproducible, you might want to suggest
changes or make them yourself if you wish. try to build starting from
0setup and then identify what's needed, raspbian-specific stuff.

**Comment by woodenshoe-wi** [comment]
I _think_ it should be reproducible.  I haven't started a build with an empty local-repositories directory and re-downloaded everything, but if I remember right when I was working on this a year ago I was avoiding using anything that was not in one of the repos.

11ff0ff stops 3builddistro-Z from installing a "Huge_Kernel" when using Raspbian kernels, If you want me to do a test x86 build could you suggest a particular version that you are familiar with building?

Regarding 01micko's [comment](https://github.com/puppylinux-woof-CE/woof-CE/issues/1#issuecomment-160388704) I get nervous partitioning and formatting but I did make a 1GB filesystem image without a swap partition.  Raspbian uses a swap file and the puppy swap file maker seems to work fine.  Without the firmware files that were in the other filesystem images it compresses down to 155K.  The only problem is that gparted won't resize the partition from a running pi because it is mounted.

This isn't a zip file, its an xz compressed filesystem image, but GitHub didn't accept it so I added a .zip to the filename.
[raspi-sd-1gb-skeleton-ext4_nojournal-8nov2016.img.xz.zip](https://github.com/puppylinux-woof-CE/woof-CE/files/579916/raspi-sd-1gb-skeleton-ext4_nojournal-8nov2016.img.xz.zip)

Do you know a packages-templates expert who could review adb4a21 ?  It would affect all builds, not just Raspbian builds.

**Comment by wdlkmpx** [comment]
I did build a puppy after merging your changes and nothing in the x86 build process was affected. puppies i'm familiar with... i built a slacko slackware 14.2 twice, so that one is working quite fine. the change your proposing looks fine too.

don't worry about paritioning and formatting, i guess that's not that hard to implement. meanwhile micko or peebee might want to upload your img (with a md5 sum) to here: http://distro.ibiblio.org/quirky/arm/sd-skeleton-images/

as that's the url in 1download..

galculator... only raspbian wheeze is using the template, all the other compat distros pick the a pet file (no templates), so it's ok. templates affect all puppies that pick that specific package from the compat distro repos, when everything seems ok, it stays for all. if there are possible issues/conflicts.. then a template for a specific distro or variant is created (dbus_slack for example (maybe))

**Comment by peabee** [comment]
> micko or peebee might want to upload your img (with a md5 sum) to here: http://distro.ibiblio.org/quirky/arm/sd-skeleton-images/

I don't have access to quirky - only puppylinux - would imagine only BarryK can upload to quirky....

**Comment by woodenshoe-wi** [comment]
If you think they are OK,
#902 #903 

At this point resizing the partition from a running pi is a problem, so the 1GB image isn't that useful right now.

**Comment by woodenshoe-wi** [comment]
I did get a Pi3.  Found out that the Pi3 wouldn't boot my wheeze based Puppy images so I started working on a Jessie build.  #908 

There is a problem with debdb2pupdb that is triggered by the main Packages file in Jessie,
```
Processing Packages-raspbian-jessie-main into a standard format...
./0setup: line 352: 28338 Segmentation fault      ${DEBDB2PUPDB} ${DISTRO_BINARY_COMPAT} ${DISTRO_COMPAT_VERSION} > /tmp/${ONE_PKGLISTS_COMPAT}temp
Processing Packages-raspbian-jessie-nonfree into a standard format...
Processing Packages-raspbian-jessie-contrib into a standard format...
Processing Packages-raspbian-jessie-firmware into a standard format...
```
I was able to work around it by making a Packages-raspbian-jessie-main in an old version of woof-CE that still used the old slow bacon debdb2pupdb.  https://www.dropbox.com/s/rot55z9zbgmlzz6/Packages-raspbian-jessie-main.xz?dl=0

I tested the build on my Pi1 Pi2 and Pi3, and it runs on all of them.  I don't have a Pi zero but it should run on that too.

Currently it is using JWM from Raspbian and jwmconfig3.  I was ending up with an empty menu trying to build without jwmconfig3, maybe its xdg_puppy again.
JWM appears to hang when shutting down, just click the desktop or scroll the scroll wheel and jwm will shut down.  I had to test the devx anyway so I compiled jwm-2.3.6 https://www.dropbox.com/sh/49y6kup5q6w88hi/AAA_HZmkwwD0QZkC8g50kRbZa?dl=0
The new JWM does fix the hanging on shutdown.

If you want to load the devx, before running sfs_load you have to open a terminal and run
` modprobe loop` and before using git to clone https repos you have to run `update-ca-certificates`

 Midori won't download files, I just use wget, and it appears /var/run is directly on the SD card so it is probably writing to the SD card way too much, but at least it is a start :grin:

**Comment by wdlkmpx** [comment]
Good job.. If raspbian jessie works fine in all rpi versions.. do you think it's ok to delete the raspian wheeze configs/build recipes?

iguleder aka dimkr actually fixed debdb2pupdb several times.. so the last version should be working ok... are you cross building or building in a pi?

i updated woof-arch/arm/build/support/debdb2pupdb .. it was never updated since the first commit, at least according to this:

https://github.com/puppylinux-woof-CE/woof-CE/commits/rationalise/woof-arch/arm/build/support/debdb2pupdb

**Comment by woodenshoe-wi** [comment]
I basically started with  `grep -v '^no' DISTRO_PKGS_SPECS-raspbian-wheezy > DISTRO_PKGS_SPECS-raspbian-jessie` and then started updating what was left.  It would still be in the git history if anyone did need to reference anything that I removed, and I don't plan on doing anything more with the wheeze build, so I think it is OK to delete them.

I do the woof-CE builds on my laptop, mostly x86 host arm target but I did try an x86_64 host arm target build and I still got the segfault with the debdb2pupdb from rationalise.

I compiled JWM natively on my Pi1 to test the devx.  I also got network swap to work so I could get around xz running out of RAM on my Pi1.

I don't know that the PPM works on the Pis but I can confirm that downloading debs from the raspbian.org website and clicking on them to install works, and clicking on the JWM pet that I made worked also.

**Comment by wdlkmpx** [comment]
peebeeeeee create 'pet_packages-jessie' in http://distro.ibiblio.org/puppylinux/arm/ and upload these files to that dir:
https://www.dropbox.com/sh/49y6kup5q6w88hi/AAA_HZmkwwD0QZkC8g50kRbZa?dl=0
also add this file (in arm/): https://github.com/puppylinux-woof-CE/woof-CE/commit/738e8df16a241f44f8125510639db8f5a5a086c5

i can confirm debdb2pupdb segfaults, so i'll look into this..

**Comment by peabee** [comment]
Done

**Comment by dimkr** [comment]
@wdlkmpx , make sure debdb2pupdb is built from the latest debdb2pupdb.c

**Comment by woodenshoe-wi** [comment]
I noticed in the git log that you moved the SD card images to the puppylinux repo.

I am able to boot the initrd that woof-CE generates with the attached 
[cmdline.txt](https://github.com/puppylinux-woof-CE/woof-CE/files/613987/cmdline.txt) and [config.txt](https://github.com/puppylinux-woof-CE/woof-CE/files/613988/config.txt)

Of course it errors out and drops me to the command line, but I am then able to resize the second partition with busybox's fdisk and the resize2fs program in the initrd.

Except for the fact that I don't know how to run fdisk non-interactively, it should be possible to make a special initrd that would automatically resize the second partition if the initrd is booted.

**Comment by wdlkmpx** [comment]
Interesting. well everything can be done, however you are (not) alone in this..

It's possible to automate everything you propose. fdisk -> http://superuser.com/questions/332252/creating-and-formating-a-partition-using-a-bash-script

And creating a special and very simple initrd init script for arm makes sense too, we have to have a technical discussion about this, although only those who have rpi's have the last word. Meanwhile @KarlGodt might want to post his ideas and code for a full install inidtrd.gz init..

I fixed debdb2pupdb and updated the binaries. I just noticed 3builddistro copies support/debdb2pupdb and support/find_cat binaries to rootfs-complete... these 2 run in the host system.. it cross builds it may not be possible to update the dbs with the ppm, and scripts calling find_cat might fail..

**Comment by woodenshoe-wi** [comment]
I'm starting another test build and I can confirm that debdb2pupdb in rationalise works and 0setup finishes successfully with no missing packages anymore. :+1: 

That is an interesting approach, piping the commands to fdisk.  I had assumed that sfdisk would be needed, which isn't in the initrd.

I was thinking of manually modifying the initrd that woof-CE makes to just resize the partition, remove the cmdline.txt, config.txt and initrd.gz files, replacing them with any backed up versions and reboot.  This modified initrd could be packed up as a pet that if installed asks if you want to resize the partition, backs up any existing cmdline.txt, config.txt and initrd.gz files, and installs the modified initrd.gz, cmdline.txt and config.txt files.

The way woof-CE makes the SD image files an initrd isn't needed to boot normally, and if someone did get the woof-CE builds running more like a frugal install with a main sfs and a save file, they could be distributed as zip files and there would be no need to resize any partition.  It might be necessary to allow for a different initrd init script for arm, but I think this should be used for booting frugally.  I think the resizing can be handled as a separate pet that is included in the package list.

**Comment by mavrothal** [comment]
If you want to script partitioning you are better off with [```sfdisk```](https://suntong.github.io/blogs/2015/12/25/use-sfdisk-to-partition-disks/) which is a command rather than menu driven program. One issue with sfdisk is that it can not handle the GUID  Partition  Table (GPT).  For scripting GPT partitioned devices you can use ```parted``` instead.

**Comment by wdlkmpx** [comment]
Well woodenshoe i'll certainly merge any changes you want to make, i'd like to help but i don't have a pi.

And don't forget to use the PPM to install software and report any issues you find, or any ideas you have..

I edited a couple PPM scripts, and well i have my own scripts to install and download deb packages, which reminds me why puppy is dead.

Many of us have written our own scripts/edited the puppy scripts for own puposes (including @rizalmartin ). And unless you're directly contributing to woofce duplication of efforts happen every single day, every single hour.

I don't see a way to address this..

In puppy everything has gone wrong.. everything that could go wrong went wrong. It's clear to me why people have forked woof or dropped it completely for other alternatives..

**Comment by mavrothal** [comment]
@wdlkmpx I can see an increasing frustration being the only one working on woof/puppy for some time now, practically with no feedback from anywhere. I can understand and sympathise. Is not fun.
Given that both Phil and Mick appear to  be occupied by real life and that peebee releases too often to appeal to a wider audience, I would urge you release a puppy of your own. Let the excellent job you do in woof be better appreciated.

**Comment by woodenshoe-wi** [comment]
@wdlkmpx Just to let you know I still do intend to help getting the Raspbian Jessie build in woof-CE working.  When I "forked" the woof-CE repo that was just because that seems to be the GitHub way of working, and I wanted to share the working Raspbian Wheeze build but didn't want to invest the time and effort to figure out why my x86 test build wasn't working.  That might be the case with other forks too, they may still be pulling in changes from upstream but unsure if their own changes might break something.

The link you gave about piping commands to fdisk was the missing piece.  I was doing some testing piping commands to busybox fdisk in the initrd on my pi and it works well on my SanDisk SD card.   When I started testing on my really cheap no brand SD cards I ran into trouble with resize2fs not actually resizing the filesystem, so I ordered two more SanDisk cards and am waiting for them to arrive before I continue testing. (I don't want to overwrite my working Raspberry Puppy :smiley: )  Busybox fdisk runs interactively, but doesn't have true menus, maybe mavrothal is thinking of cfdisk?  The example for sfdisk was still piping commands, the format of the commands was just different than busybox's fdisk.

Even BarryK had some trouble with cross builds in WoofQ http://murga-linux.com/puppy/viewtopic.php?p=925090#925090 

Don't worry about not being able to help test Pi builds, I can handle that, the help I need is getting a better idea of how things are done in woof-CE going forward.  For example @peabee suggested moving the "z-hacks" into "package-templates" for "just that distro" #[878](https://github.com/puppylinux-woof-CE/woof-CE/issues/878#issuecomment-248837181), which does sound like a good idea, and I would volunteer to move the "z-hacks" I use in the Raspbian Jessie build, but I don't know where the "package-templates" for "just that distro" is located.  I know of a main "package-templates" that affects **all** builds, but some people get nervous if you try to make changes there.  Should there be a "package-templates" in woof-distro for each distro?  I would be willing to try to implement that if that is the direction woof-CE wants to go, but if that is not the direction woof-CE is going it would just be a waste of time.

I think making a puppy release takes a lot of work to fix all the bugs the testers find.  I don't know first hand, and I don't _want_ to know first hand, I just want to fix the build in woof-CE and if someone else wants to make a release of it I wish them all the best.  I think woof-CE deserves to be worked on in it's own right too, its not just a disposable tool to be thrown away after a release is made, but it is impossible to expect everyone who makes a change to test every possible build that they are not familiar with to make sure they haven't broken anything.  Maybe if it were possible to somehow untangle woof-CE from the individual builds people could work on the individual builds without worrying about breaking something in an unrelated build.  I was surprised by how many things I had to change in the main woof-CE scripts, from changing how the image file was mounted, to editing xorg.conf to get wheeze to boot to desktop.  Missing a quotation mark or something could break the script.

I know this is getting long, but I probably won't be working on this until my SD cards arrive and I wanted to leave a "status report".

**Comment by wdlkmpx** [comment]
i must admit that i'm "tired", "worn out", "etc". i've been certainly
thinking about a test release, but i stopped building and checking my
precise updates as i kept editing scripts, huge edits right now. but
i'll certainly continue with it and then i'll try to make a "debian
stretch", meanwhile peebee is doing a great job with his releases that
truly reflect the current state of woof, but of course partially
50/50.

ffmpeg returns with debian stretch so that's the only worthy debian
release in a long time. for the retro stuff there is the ubuntu
puppies where the key is precise when no work is done to create proper
nvidia/amd packages for later versions which involve downgrading xorg
for the most retro propietary drivers. There is a slackware doc that
explains how to do that with slackware 14.12... it's possible to do
everything with slackware even for the most retro video hardware
(1999+).

Talking about Stretch.. woodenshoe, how about start upgrading from
raspian jessie to raspbian stretch... i see this:
http://archive.raspbian.org/raspbian/dists/stretch/

But it may be too unstable... but it's worthy hehe... stretch will
land in the first months of 2017..

**Comment by woodenshoe-wi** [comment]
I still have to add some more dependencies I found while I was trying to compile the latest gtkdialog, waiting on the SD cards, and I want to figure out why I get an empty menu using the jwmconfig from woof-CE.

Once I feel DISTRO_PKGS_SPECS-raspbian-jessie is finished I'll try adding support for stretch, just don't delete jessie until stretch is stable :wink: 

I think I will have time to work on this stuff until April, but over the summer I may be to busy until November or so.

**Comment by wdlkmpx** [comment]
The z-hacks are fixup files and scripts.. a "schema" file is
missing... it goes in a z hack. Something like that i think. When it
comes to xorg.conf, etc... probably it's better to put those files in
rootfs-skeleton?.... have you tried the new xorg.conf for arm?

The packages-templates dir is only to preprocess/post process compat
distro packages. Not for pet pkgs - z hacks.. although it's easy to
place them there: 'packages-templates/z_distro_version_hacks' and make
2createpackages copy that dir to packages-distro... although
support/distro_version makes sense too..

**Comment by wdlkmpx** [comment]
uhmm i think the jwmconfig is basically the reason for the
complication in 3builddistro with the dialog.

the woofce jwmconfig requires jwm 2.3.2 compiled with the jwm.petbuild
in the petbuilds repo or something like that. to configure it and
produce a wide variety of themes and looks, it requires ptheme.
although the design is top notch..

so, for newer jwm version, stick with the jwm config from the repos..
which i have to edit as well fix and add keybindings..

**Comment by woodenshoe-wi** [comment]
Did you mean 9c06459 ?
The problem was with one line
```
#	DefaultDepth 24   # Xorg from Raspbian won't start if uncommented.
```
but I didn't want to change it in all builds because I didn't know what I might break.
Everything is working fine now with xorg.

I haven't looked inside one of the "z-hack" pets, I just know that DISTRO_PKGS_SPECS-raspbian-jessie includes z_geany_bacon_hack, zz_ubuntu_precise_fixup, z_lowresource_hack and z_raspi_fixes_hack.  If they are scripts they might be easier to edit if they where somewhere in woof-CE instead of in a repo.  I would like to automatically run `update-ca-certificates` but don't want to make a pet that someone else has to upload to a repo just to run one command on first boot.

Some of the templates in packages-templates add scripts or .desktop files I think.  If it is text based and missing from that particular package it can be added with a template.  Not for binary files as it must be architecture independent.  The only problem is if version x of some package is missing its .desktop file and you add it in a template and then version y comes out and the problem is fixed the template would still be hanging around in woof-CE because nobody would want to delete it because that might break something...  If the "hackish" templates could be somewhere in woof-distro they would automatically go away when the distro version that used them went away.

Back when I was working on the wheeze build in testing I had the same problem with the empty menu and could fix it by recompiling the latest xdg_puppy, but then the problem went away by its self.  I wanted to investigate exactly what is causing the problem, but for that I need to do a build that has the problem.
```
I found back some of my notes...
...
xdg_puppy-0.7.6-15-binaries_only-armv6.pet
is too old to handle the "MENHEIGHT" option in
/usr/sbin/fixmenus
...
```

**Comment by wdlkmpx** [comment]
This is the current xorg.conf for arm: https://github.com/puppylinux-woof-CE/woof-CE/blob/rationalise/woof-code/rootfs-skeleton/etc/X11/xorg.conf-auto-armsystem .. it can be tweaked a bit.

The packages-templates are only applied when a compat pkg is being processed, so code must be added for the script to use z hacks for a specific distro-version. Z hacks in packages-templates must have distro_version in the name.

it can also in a create a subdirectory in woof-distro/arch/distro/version/ -> 'hacks' or something like that and put all those files there.

Well, the z_hacks are basically missing files, i guess there are also a couple scripts here and there, but i'm not sure.
This: z_raspi_fixes_hack might be of interest, unpack it and see what it has to offer..

Just unpack all the z hacks pet pkgs and you will see random files the builders added..

**Comment by wdlkmpx** [comment]
After commit e243accb62ab1a5657df91d6c171896658d4a752 you can place all your custom files and scripts including a pinstall.sh script in ```packages-templates/z_raspbian-jessie_arm``` (create that dir)

Well i can see everything can still be a pet pkg, including the huge kernels... (3builddistro would move the kernel and sfs files to the build/ dir.. about 60 lines would be gone...)

**Comment by woodenshoe-wi** [comment]
The point I was trying to make about  9c06459 was that I added xorg.conf-auto-armsystem because there was one line in xorg.conf-auto-pc that kept Xorg from Raspbian from running and I was afraid that changing it in xorg.conf-auto-pc might break other puppy builds that used an older Xorg.  Since your commit cd3e62c xorg.conf-auto-pc and xorg.conf-auto-armsystem are functionally identical and 9c06459 can be reverted.

I looked in z_geany_bacon_hack-1.5 and found a pet.specs file, a sym link, a bash script and two config files.

In z_lowresource_hack-20120707 I found a pet.specs file, a pinstall.sh script that has one line disabling system logging and one "if statement" regarding the chromium-browser, and an init.d shell script for cleaning up after the chromium-browser.

In z_raspi_fixes_hack-20120709 I found a pet.specs file, a pinstall.sh script with a one line sed substitution on etc/rc.d/MODULESCONFIG, and an init.d script to start alsa.

In zz_ubuntu_precise_fixup-20120522 I found a pet.specs file and that one gschema.xml file you probably were using as an example of what goes into z-hacks.

I finally found cross-build-fix in woof-code/rootfs-skeleton/etc/rc.d/rc.update which is where I would have to add `update-ca-certificates` I guess.


When I got frustrated and quit trying to get my changes merged last December, it wasn't because GitHub was too difficult to use, or because the scripts weren't documented well enough.  I can always just read the scripts to figure out what they do.  But as I was getting my Raspbian build working I was changing files all over woof-CE, not just in woof-distro/arm/raspbian.  Adding xorg.conf-auto-armsystem and now having to edit rc.update are examples of where I get nervous because I don't want to break something that I can't even test, I don't know what version of Xorg some retro build is using, and maybe not all builds include  ca-certificates.

If new would-be woof-CE developers could put their changes in  woof-distro/${TARGETARCH}/${COMPATDISTRO}/${COMPATVERSION}/ packages-templates,rootfs-packages,rootfs-skeleton and a patches directory that could hold patch files to change just about anything in woof-CE without worrying about breaking something in another build, merge2out could layer the distro specific changes on top of the existing woof-CE.  This would give distro developers complete control over their builds, but I don't know if this would cause problems because woof-CE could end up forked right inside woof-CE.  It might still be easier to find the changes if they weren't spread out all over GitHub, but I don't know if this is something the woof-CE developers want to maintain.

I would be willing to try to implement this, I'm not asking anyone else to do extra work, I just want a go, no-go decision before I would try to make a change this big.

With e243acc I could merge all the z-hack pets into packages-templates/z_raspbian-jessie_arm, but it turns out I don't have to change anything in them, so it's a matter of preference, clutter up packages-templates with z_raspbian-jessie_arm, complicate woof-CE with woof-distro/arm/raspbian/jessie/packages-templates/z_raspbian-jessie_arm, or just leave them as they are?  I guess I also have trouble making decisions...

**Comment by wdlkmpx** [comment]
Well we all are full of doubts, but as i said, just do what you think
is best. The main problem here is the lack of communication. To
further discuss this maybe the forum is the most appropriate place,
but all topics about woofce are a failure due to many factors. I
really don't like the murga forum.

well everything is clear to me now.. when it comes to x86 what defines
a distro version such as in debian is basically xorg+kernel. A puppy
can override both and provide a different xorg/kernel version
targetting a specific hardware range. And using the same base, so a
semi retro puppy (i686) can still happen using the latest release. but
then again i don't know whether to start with debian stretch or devuan
ascii.. devuan is the logical choice. I'll delete precise and
slackware 13.37...

**Comment by mavrothal** [comment]
> If new would-be woof-CE developers could put their changes in woof-distro/${TARGETARCH}/${COMPATDISTRO}/${COMPATVERSION}/ packages-templates,rootfs-packages,rootfs-skeleton and a patches directory that could hold patch files to change just about anything in woof-CE without worrying about breaking something in another build, merge2out could layer the distro specific changes on top of the existing woof-CE. This would give distro developers complete control over their builds, but I don't know if this would cause problems because woof-CE could end up forked right inside woof-CE. It might still be easier to find the changes if they weren't spread out all over GitHub, but I don't know if this is something the woof-CE developers want to maintain.

I would think that would be better to keep the ZHACKSDIR  for **new** arch-specific files and have a ZPATCHDIR for arch-specific patches of existing woof files. This would hopefully  make clear which scripts of the woof-code are problematic for the specific arch and allow everyone to pitch in trying to consolidate them. 
The downside is that changes in the "main" woof files may break the patches but hopefully because the incorporate them. If we could have a [git hook](https://git-scm.com/book/en/v2/Customizing-Git-Git-Hooks) to run a script and flag changes in the main files that break patches, it may help in this case. The script would go through the files changed in the commit (by ```git show --pretty="" --name-only <HEAD hash>``` or something similar specific to the hook mechanism) and look if they are patched in ZPATHDIR for those. If ```patch --dry-run -p1 < patch_of_changed_file``` then fails could block the commit ( 😮  maybe too drastic) or amend the commit message with \"This commit breaks patch fix/my/build5\" and write a warning to a log. 
The problem would be in cases where files are changed through the github web interface, but I do not think that anything can be done about it.

Alternatively, a "demanding" ARCH could have its own branch but it may be more difficult to merge them down the road and/or take advantage of improvements in each branch.
Hopefully other people will come up with better ideas. 
That's just my 2c while sipping my coffee 👅 .

**Comment by wdlkmpx** [comment]
for raspbian stretch, you might want to just copy some or most entries
(libs, ffmpeg, etc) from debian stretch x86 and adapt them to your arm
build

i'm updating the pkgs specs... there will be just a few changes before
stretch becomes the stable release.  most stuff is already freezed i
think.. it will probably take one or two weeks to get this working if
i don't switch to do other things. so start around december 15 ...

**Comment by woodenshoe-wi** [comment]
As far as I know some of the z-hack pets have pinstall.sh scripts that have to be run by 3builddistro-Z and others just add missing files or config files that are not appropriate for rootfs-skeleton.  Anything that needs a pinstall.sh to be run would have to be in ZHACKSDIR, but I was hoping that the missing files and config files could go in woof-distro/${TARGETARCH}/${COMPATDISTRO}/${COMPATVERSION}/rootfs-skeleton
That way if someone was doing a build with very new programs or very retro programs they could override any config files in rootfs-skeleton that were causing problems without affecting other builds that use more normal versions of programs.

The same thing with woof-distro/${TARGETARCH}/${COMPATDISTRO}/${COMPATVERSION}/ packages-templates
Before all the extra documentation was cleaned up in 3builddistro, I had made a pair of templates to clean up the Raspbian kernel packages, but no other builds will ever use Raspbian kernels...  When I change a template to make a program work right in Raspbian (galculator, it has been merged to rationalise), I don't know if the template will work right on Slackware packages, I've only tested it on Raspbian packages.

The idea of having woof-distro/${TARGETARCH}/${COMPATDISTRO}/${COMPATVERSION}/rootfs-packages was that if there were any script based gtkdialog packages that were in some way specific to that kind of build they wouldn't clutter up rootfs-packages.  Any binaries should still be made into pets, you can't edit a binary in git anyway.  Maybe ZHACKSDIR could go there, then when an old distro is deleted everything gets cleaned up automatically.

The idea of woof-distro/${TARGETARCH}/${COMPATDISTRO}/${COMPATVERSION}/patches was either temporary patches from someone working on getting puppy to run on their new toy (like me and my Pis) who doesn't feel confident enough to ask the changes be made in the main woof-CE scripts (one of the woof-CE devs could help getting the changes made properly to the woof-CE scripts and the patches could go away) or an attempt at moving distro specific parts of the woof-CE scripts out of the main woof-CE scripts.  (Like writing to an SD card image file).  There could be comment "hooks" added to the main woof-CE scripts to give patch something to "grab on to" or maybe it would be better to rewrite them as helper scripts and leave them in the main woof-CE.
Either way I think it should be the responsibility of whoever is using the patches to either always use them with a specific version of woof-CE (like "main" or some tag) and try to get them merged or fix them when any changes break them.  I don't think woof-CE should try not to break them unless they are using some kind of "hook" system agreed to by the woof-CE devs.

These are just my ideas as a newcomer would be woof-CE contributor who can figure out GitHub and the woof-CE scripts enough to get my builds working, but am not enough of an expert to be confident I won't break something.

I think merge2out could do all of the work, and I would be willing to try to implement this, but I don't want to end up making a mess that is hard to maintain.

**Comment by wdlkmpx** [comment]
@woodenshowe-wi .. after commit 7ff90367129a49f45e764abdba4121235b57cfe1 you can do that. the important thing is that your changes end up in woofce one way or another..

**Comment by woodenshoe-wi** [comment]
I think it is a little more complicated than that, it is easier to understand templates and packages as complete units, but rootfs-skeleton needs to be an overlay.

I issued #910, haven't done any full builds but tested merge2out several times.

I didn't implement any patch support but I remember reading about a patch generator for woof-CE and would want to be compatible with it.

My SD cards are supposed to come by Saturday, so then I will be doing more builds.

**Comment by wdlkmpx** [comment]
if you want to create templates that add desktop files, etc. then it's
safe to use/edit the 'standard' templates as most of these are
outdated...

well regarding stretch there are a couple libs required by a bunch of
apps/libs that are nowhere to be found. strange. i guess this won't
happen again starting from january 2017. meanwhile i have something
that boots to desktop but i need to compile jwm and rox. to avoid
compiling apps i'm also trying  to use as many apps as possible from
the compat distro repos - updated, bigger, but no uploads to pet
repos. what a relief. gftp is missing right now. no gnome-mplayer...
what is this!?

so for templates, i'll probably edit some of them

**Comment by woodenshoe-wi** [comment]
I don't know if this will help, but while working on my Raspbian build I discovered the Content-armhf.gz file.  I downloaded it and can search for missing libs by running
```
gunzip -c Contents-armhf.gz | grep -F '<missing file name>'
```
At the US Debian mirror I found a bunch of Contents-some_arch.gz files
http://ftp.us.debian.org/debian/dists/stretch/main/

**Comment by woodenshoe-wi** [comment]
I compiled jwmxdgmenu-0.7.6 and gtkdialog-0.8.4  
https://www.dropbox.com/sh/49y6kup5q6w88hi/AAA_HZmkwwD0QZkC8g50kRbZa?dl=0

jwmxdgmenu fixes the empty menu problem with the jwm_config from woof-CE, and gtkdialog
lets me run ptheme, but I still get a lot of GLib-GObject and GLib-GIO warnings.

**Comment by wdlkmpx** [comment]
those files will be in the repos in a few hours ... peebeee

well, i can't play most vids in debian stretch (some do play hehe), ffplay segfaults. if this happens even in january 25 2017... then i'm missing something or it's because i haven't downgraded xorg yet. stretch is being updated right now, with packages being added/updated.

rox and jwm are in debian stretch, but i know rox is abandonware that is being fixed very unofficialy. the jwm 2.3.6 from stretch works fine and all these little things about menus just get in the way having to compile them everytime, i haven't and i want to get rid of xenial and tahr repos... it's a lot of work to get a puppy working fine and really not fulfilling, you become a slave for everything you want to improve. that's why i think there should be at least 2 developers involved for each compat distro-version, the main developer and the pupil, and one or two dedicated testers before an iso is released... so to get at least something working for ourselves, it's ok.

**Comment by woodenshoe-wi** [comment]
I've been working on an sd-resize pet.  I want to test it in one more build, but it seems to be working.  I also finally got my kernel templates to clean out the config and System.map files from /boot.

I was hoping that recompiling gtkdialog against the version of gtk from Raspbian would get rid of the warnings, but it didn't.

Now I'm wondering, I have the dev files for GTK 2 in the devx, but some of the programs from Raspbian needed GTK 3, so the run time libraries for GTK 3 are in the main image file also.

Should I have included the dev files for GTK 3 instead of GTK 2?

Also, are gtkdialog4 and gtkdialog3 still needed or does everything use one gtkdialog program with symlinks?

I don't want to commit to the workload of being a developer, but as long as I am not causing trouble :wink: I'll gladly share any work I'm doing anyway.  I like Puppy better that the official Raspbian images.

I think I can figure out any of the Pi specific issues, but do you know where I could get help about the things that are the same in all woof-CE builds?  The old Raspbian wheeze package lists hadn't been touched since woof-CE was started, and I haven't been keeping up with the new developments in woof-CE, so I have questions like

Is Puppy using GTK 2 or GTK 3?

Should I be using jwm_config from woof-CE or jwmconfig3 from the repo?

If I was trying to make a basic minimal version which packages are basic requirements?

**Comment by peabee** [comment]
> those files will be in the repos in a few hours ... peebeee

Uploaded
If this is going to be needed more in the future then somebody else should ask @01micko for access to ibiblio......

**Comment by woodenshoe-wi** [comment]
I finally got my sd-resize pet working.

I had been copying a script to check for missing dependencies to the SD card after writing the image file to it, but before booting it for the first time.  I didn't discover that resize2fs refuses to resize a partition that hasn't been mounted at least once until I started testing the pet.

The pet is made from the armv6 programs from initrd_progs-20161109-static.tar.xz  
and I tested it on my Pi1 and Pi3.

@peabee would you please upload just 3 more files to ibiblio for me?  
sd-resize-0.1-armv6.pet should go in arm/pet_packages-armv6  
raspi-sd-1gb-skeleton-ext4_nojournal-8nov2016.img.xz and  
raspi-sd-1gb-skeleton-ext4_nojournal-8nov2016.img.xz.md5.txt  
should go in arm/sd-skeleton-images
https://www.dropbox.com/sh/2y66rmgzfgh8e1g/AAD75yOP4_w61xQHlLgNZj34a?dl=0

If anyone else was interested in doing Pi builds I would be willing to try to compile more stuff, but if I am the only one doing builds I want to keep it to just the basics in DISTRO_PKGS_SPECS-raspbian-jessie.

**Comment by peabee** [comment]
'tis done

**Comment by wdlkmpx** [comment]
It's ok to have both gtk2 and gkt3. Projects are dropping gtk2, having development files for both is ok. To make a config that can be easily used in x86_64 and arm.. i'm simplifying this.

Debian has most apps, including yad
https://packages.debian.org/stretch/yad

I installed pcmanfm with the ppm and works quite well. My advice is use whatever is available in debian at any cost.. except udev and other apps from the systemd source pkg. maybe devuan ascii has compatible and fixed pkgs..

**Comment by woodenshoe-wi** [comment]
@peabee thanks for uploading the files, but could you add the pet.specs for  
sd-resize-0.1-armv6.pet to the Packages-puppy-armv6-official file?

./1download can't find it.

**Comment by wdlkmpx** [comment]
already updated in woof https://github.com/puppylinux-woof-CE/woof-CE/commit/7ef0238554017803fb5e0f034022dde7136bfda6 .. there is the same file in the pet repos that needs to be updated, but it will work nonetheless (you will see letters in red, probably) 

also updated the galculator template https://github.com/puppylinux-woof-CE/woof-CE/commit/d1aea883617aed3e3ef82e86e5a60ee58941fc88

i wonder how other people work and how they deal with rox. 

it really is a challenge to do this in rox using only the keyboard, it takes minutes. this usually takes me seconds when i'm using pcmanfm having tab pages and keyboard shortcurts for basically everything useful, as i have to test code/templates and copy/paste files between dirs. it does affect my workflow severely, i just can't fool myself anymore... i like rox, i like rox, you get signal, all your base are belong to us..

**Comment by woodenshoe-wi** [comment]
I'm doing a build now.  I had started earlier this evening but then remembered that I had to do something about the galculator template, and when I looked on GitHub to see what I had changed the first time I saw you had already fixed it. :+1: 

I have absolutely no experience using rox without a mouse, but do have a little experience using my Raspberry Pi without a working keyboard...  :grinning: looking for the letters I needed to paste together commands with the mouse I felt like I was writing a ransom note.

The only time I can think of having to use only the keyboard is using ssh, but that is command line only, except over the home network I had rox working with X forwarding.

**Comment by woodenshoe-wi** [comment]
OK, I finished the test build and everything looks good.

It's definitely not release quality, but I wouldn't say that the Raspberry Pi build is broken any more. :grinning: 

The 1GB image file should write to a SD card four times faster than the 4GB image file did, the problem with the ca-certificates has been fixed with a template in raspbian/jessie, and jwmxdgmenu and gtkdialog are now what I think are the latest versions.

The problem with Midori not downloading files seems to have gone away by itself. :roll_eyes: 

#915

**Comment by woodenshoe-wi** [comment]
I've been working on getting the builtin wifi on my Pi3 working.  I looked everywhere I could think of in the Raspbian repos for the wifi firmware but couldn't find it so I made a pet with the firmware from https://github.com/RPi-Distro/firmware-nonfree

This is only the firmware for the Pi3, it doesn't have any other brcm/Broadcom firmware in it.  
https://www.dropbox.com/sh/2y66rmgzfgh8e1g/AAD75yOP4_w61xQHlLgNZj34a?dl=0

I don't know if it should go in pet_packages-armv6 because it is only used in arm builds, or if it should go in pet_packages-noarch with the rest of the firmware?

**Comment by woodenshoe-wi** [comment]
I started working on a build based on Raspbian stretch, #940 I think I have compiling working now.

@wdlkmpx do you know where I can find a petbuild or at least the configure flags for eudev?  And is there a specific version I should use?

I was able to compile the latest version on my pi3 and after adding the mail group `make check` passes.  I would like to recompile it on my pi1 the same way the pet in your Debian stretch build was compiled.

**Comment by wdlkmpx** [comment]
I used an edited PKGBUILD (Arch), i have the file somewhere. i'll post it later. it's for eudev 3.2.1,

The network interfaces have weird names (enp2s1). Apparently i'm missing these files https://github.com/gentoo/eudev/tree/master/rule_generator also here http://anduin.linuxfromscratch.org/LFS/udev-lfs-20140408.tar.bz2

**Comment by wdlkmpx** [comment]
This is the updated script i use to produce a complete eudev pet pkg: http://pastebin.com/raw/JnWd3LVK

run this way:
./eudev 

when it's a finished you'll see a pet pkg ready to go in the pet repos.

a new feature in 2createpackages is that you can "fake" the installation of problematic apps by specifying exe>null,dev>null,doc>null,nls>null

see commit: 8705f6b81f0e0a805dab2415f237840ea457f774  

from now on you might want to work in testing as both branches are even..

**Comment by wdlkmpx** [comment]
oops i think i have to redo the last commits i pushed to rationalise. apparently they got lost

**Comment by wdlkmpx** [comment]
hmm no. the commits are there. they're just hidden somewhere in the log. strange, as the commit date is 2017-01-11

**Comment by woodenshoe-wi** [comment]
Success! :grin:  
I compiled the eudev pet on my pi1, did a test build with with the eudev pet and a fake install of systemd-udev, and now my wifi works properly again.  No more strange interface  names...

Pet compiled in Raspbian stretch based puppy (raspup-7.0.1)  
https://www.dropbox.com/sh/rgrfxxevokj8994/AACE-xx5zBbeIEQQ8OlbysxZa?dl=0  
I think it should go in arm/pet_packages-stretch/ with an entry in Packages-puppy-stretch-official

**Comment by peabee** [comment]
> I think it should go in arm/pet_packages-stretch/ with an entry in Packages-puppy-stretch-official

These new items created.....

**Comment by woodenshoe-wi** [comment]
I have another pet for arm/pet_packages-stretch /  Packages-puppy-stretch-official

I have been trying woof builds on my pi3 connected to a computer over Ethernet with the files on the computer's hard drive.  I discovered the old disktype would segfault when reading the empty SD image file so I compiled the latest version from http://distro.ibiblio.org/puppylinux/sources/

The original disktype was compiled statically, but I compiled the new one on my pi1 as a regular dynamically linked program.  I don't know where it would be used other than during woof builds.  If it needs to be compiled statically maybe it could be added to initrd-progs?  Otherwise if it is OK to use the dynamically linked version, @peabee if you wouldn't mind, could you upload please?
https://www.dropbox.com/sh/rgrfxxevokj8994/AACE-xx5zBbeIEQQ8OlbysxZa?dl=0

**Comment by peabee** [comment]
Done.

**Comment by 01micko** [comment]
Ok, since I created this issue all those years ago can we close it? Of course comments are still supported whether it is closed or not.

**Comment by woodenshoe-wi** [comment]
It probably is still broken in master, but that can't be helped because the base distro is always changing and the updated DISTRO_PKGS_SPECS will be merged into testing because that is the branch that I do my builds from.

I think the stretch build in testing should work, although I haven't done a jessie build recently.

I would say since testing is the default branch, if having a working build in testing is good enough for you, go ahead and close it.  👍

**Comment by 01micko** [comment]
Ok thanks @woodenshoe-wi 

NB: testing will eventually be merged to master

**Comment by woodenshoe-wi** [comment]
Maybe this is a newbie question, but I'll ask it anyway.

It's just theoretical right now, but if Raspbian stretch is not stable by the time testing is merged to master would it be ok to fix DISTRO_PKGS_SPECS directly in master, or should changes only be made in testing?
